manifests/master/sysconfig.pp

```diff
@@ -65,16 +69,29 @@
 
   file { $_java_temp_dir:
     ensure => 'directory',
-    owner  => 'puppet',
-    group  => 'puppet',
+    owner  => $user,
+    group  => $group,
     mode   => '0750'
   }
-
-  file { '/etc/sysconfig/puppetserver':
+ if ($server_distribution == 'PC1') {
+  file { "/etc/sysconfig/${service}":
     owner   => 'root',
     group   => 'root',
     mode    => '0640',
     content => template('pupmod/etc/sysconfig/puppetserver.erb'),
-    notify  => Service[$::pupmod::master::service]
+    notify  => Service[$service]
+  }
+ }
+ if ($server_distribution == 'PE') {
+  pe_ini_subsetting { "pupmod::master::sysconfig::javatempdir":
```

Comment:
Is this an augeas provider? There's a ticket around here somewhere for using that more thoroughly...